services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/IntersectionComputer.java

```diff
@@ -0,0 +1,132 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Computes intersection counters for flow group.
+ */
+public class IntersectionComputer {
+    private Set<Edge> mainFlowEdges;
+    private Set<SwitchId> mainFlowSwitches;
+
+    // FlowId, List<Edge>
+    private Map<String, List<Edge>> otherEdges;
+
+    /**
+     * Construct intersection counter for current flow group and target flow.
+     *
+     * @param mainFlowId the flow id to find overlapping with.
+     * @param segments flow segments in group.
+     */
+    public IntersectionComputer(String mainFlowId, Collection<FlowSegment> segments) {
+        List<Edge> mainFlow = segments.stream()
+                .filter(e -> e.getFlowId().equals(mainFlowId))
+                .map(Edge::fromFlowSegment)
+                .collect(Collectors.toList());
+
+        mainFlowEdges = new HashSet<>(mainFlow);
+        mainFlowSwitches = mainFlow.stream()
+                .flatMap(e -> Stream.of(e.getSrcSwitch(), e.getDestSwitch()))
+                .collect(Collectors.toSet());
+
+        otherEdges = segments.stream()
+                .filter(e -> !e.getFlowId().equals(mainFlowId))
+                .collect(Collectors.groupingBy(
+                        FlowSegment::getFlowId,
+                        Collectors.mapping(Edge::fromFlowSegment, Collectors.toList())
+                ));
+    }
+
+    /**
+     * Returns {@link OverlappingSegmentsStats} with all other paths in group.
+     *
+     * @return {@link OverlappingSegmentsStats} instance.
+     */
+    public OverlappingSegmentsStats getOverlappingStats() {
+        return computeIntersectionCounters(
+                otherEdges.values().stream().flatMap(Collection::stream).collect(Collectors.toList()));
+    }
+
+    /**
+     * Returns {@link OverlappingSegmentsStats} with {@param flowId} flow path.
+     *
+     * @return {@link OverlappingSegmentsStats} instance.
+     */
+    public OverlappingSegmentsStats getOverlappingStats(String flowId) {
+        return computeIntersectionCounters(otherEdges.getOrDefault(flowId, Collections.emptyList()));
+    }
+
+    OverlappingSegmentsStats computeIntersectionCounters(List<Edge> otherEdges) {
+        Set<SwitchId> switches = new HashSet<>();
+        Set<Edge> edges = new HashSet<>();
+        for (Edge edge : otherEdges) {
+            switches.add(edge.getSrcSwitch());
+            switches.add(edge.getDestSwitch());
+            edges.add(edge);
+        }
+
+        int edgesOverlap = Sets.intersection(edges, mainFlowEdges).size();
+        int switchesOverlap = Sets.intersection(switches, mainFlowSwitches).size();
+        return new OverlappingSegmentsStats(edgesOverlap,
+                switchesOverlap,
+                percent(edgesOverlap, mainFlowEdges.size()),
+                percent(switchesOverlap, mainFlowSwitches.size()));
+    }
+
+    private int percent(int n, int from) {
```

Comment:
what if it's zero?

Reply:
For current logic from couldn't be zero. But if it is - floating point division will give NaN, and external cast to int gives zero. Added a test for this situation